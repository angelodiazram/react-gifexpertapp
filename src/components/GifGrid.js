import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { getGift } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    // const [images, setImages] = useState([]);
    const {data, loading} = useFetchGifs(category);

    // useEffect(()=>{
    //     getGift(category)
    //         .then(imgs => setImages(imgs));
    // }, [category]);




    return (
        <>
            <h3>{category}</h3>

            {loading && <p>loading</p>}
                {/* operador ternario reducido*/}
            <div className="card-grid">
                
                {
                    data.map( img =>(
                        <GifGridItem 
                            key={img.id}
                            {...img}
                        /> 
                    ))
                }    
                
            </div>
        </>
    );
};