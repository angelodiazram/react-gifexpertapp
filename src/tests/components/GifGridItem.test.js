import { React } from "react";
import '@testing-library/jest-dom';
import { shallow } from "enzyme";

import { GifGridItem } from "../../components/GifGridItem";

describe('pruebas en el componente <GifGridItem />',()=>{

    const title = 'Soy el titulo';
    const url = 'https://localhost.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url}/>);
    
    test('debe de mostrar <GifGridItem /> correctamente',()=>{
        

        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de tener un parrafo con el title',()=>{
        
        const p = wrapper.find('p');
        console.log(p.html());
        expect( p.text().trim()).toBe(title);

    });

    test('debe tener al imagen igual al url y alt de los props',()=>{
        
        const img = wrapper.find('img');
        // console.log(img.props().src);

        expect(img.props().src).toBe(url);
        expect(img.props().alt).toBe(title);
    });

    test('el div debe de tener la clase css animate__fadeIn',()=>{
        
        const div = wrapper.find('.card');
        const className = div.props().className;
        console.log(className);
        // console.log(div.props().className);

        expect(className.includes('animate__fadeIn')).toBe(true);
    });

}); 