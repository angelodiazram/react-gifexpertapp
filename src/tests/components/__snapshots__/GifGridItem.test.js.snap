// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente <GifGridItem /> debe de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Soy el titulo"
    src="https://localhost.jpg"
  />
  <p>
    Soy el titulo
  </p>
</div>
`;
